pkg/expressions/json_templates_test.go

```diff
@@ -284,6 +284,166 @@ value13: | # This is a string
+				require.Contains(t, err.Error(), "literal not terminated")
+			},
+		},
+		{
+			name:         "standard delimiters with $~~ inside expression",
+			jsonTemplate: `{ "AString": "${{ \"This string contains literal \" + quote(\"$~~\") + \" inside it\" }}" }`,
+			assertions: func(t *testing.T, jsonOutput []byte, err error) {
+				require.NoError(t, err)
+				parsed := testStruct{}
+				require.NoError(t, json.Unmarshal(jsonOutput, &parsed))
+				require.Equal(t, "This string contains literal \"$~~\" inside it", parsed.AString)
+			},
+		},
+		{
+			name:         "alternative delimiters with ${{ inside expression",
+			jsonTemplate: `{ "AString": "$~~ \"This string contains literal \" + quote(\"${{\") + \" inside it\" ~~" }`,
+			assertions: func(t *testing.T, jsonOutput []byte, err error) {
+				require.NoError(t, err)
+				parsed := testStruct{}
+				require.NoError(t, json.Unmarshal(jsonOutput, &parsed))
+				require.Equal(t, "This string contains literal \"${{\" inside it", parsed.AString)
+			},
+		},
+		{
+			name:         "mixed delimiters - standard comes first",
+			jsonTemplate: `{ "AString": "${{ aString }}", "AnotherString": "$~~ quote(\"{{.tag}}\") ~~" }`,
+			assertions: func(t *testing.T, jsonOutput []byte, err error) {
+				require.NoError(t, err)
+				var parsed map[string]string
+				require.NoError(t, json.Unmarshal(jsonOutput, &parsed))
+				require.Equal(t, "hello", parsed["AString"])
+				require.Equal(t, "{{.tag}}", parsed["AnotherString"])
+			},
+		},
+		{
+			name:         "mixed delimiters - alternative comes first",
+			jsonTemplate: `{ "AString": "$~~ quote(\"{{.tag}}\") ~~", "AnotherString": "${{ aString }}" }`,
+			assertions: func(t *testing.T, jsonOutput []byte, err error) {
+				require.NoError(t, err)
+				var parsed map[string]string
+				require.NoError(t, json.Unmarshal(jsonOutput, &parsed))
+				require.Equal(t, "{{.tag}}", parsed["AString"])
+				require.Equal(t, "hello", parsed["AnotherString"])
+			},
+		},
+		{
+			name:         "both delimiters present but only standard is used",
+			jsonTemplate: `{ "AString": "foo ${{ aString }} and also $~~ ignored ~~" }`,
+			assertions: func(t *testing.T, jsonOutput []byte, err error) {
+				require.NoError(t, err)
+				parsed := testStruct{}
+				require.NoError(t, json.Unmarshal(jsonOutput, &parsed))
+				require.Equal(t, "foo hello and also $~~ ignored ~~", parsed.AString)
+			},
+		},
+		{
+			name:         "overlapping delimiters - alternative comes first by position",
+			jsonTemplate: `{ "AString": "$~~ 'test' ~~ and ${{ aString }}" }`,
+			assertions: func(t *testing.T, jsonOutput []byte, err error) {
+				require.NoError(t, err)
+				parsed := testStruct{}
+				require.NoError(t, json.Unmarshal(jsonOutput, &parsed))
+				// $~~ appears first (index 0), so alternative delimiters are used for entire value
+				// This means ${{ is treated as literal text, not as a delimiter
+				require.Equal(t, "test and ${{ aString }}", parsed.AString)
+			},
+		},
```

Comment:
The results you are expecting in these two cases are not what I think anyone would intuitively expect them to be. `${{ expr1 }}-$~~ expr2 ~~` should have _both_ expressions evaluated.